Stratis.Bitcoin.Features.Miner/PowMining.cs

```diff
@@ -155,25 +155,28 @@ public List<uint256> GenerateBlocks(ReserveScript reserveScript, ulong generate,
                     if (newChain.ChainWork <= chainTip.ChainWork)
                         continue;
 
-                    this.chain.SetTip(newChain);
+                    var blockResult = new BlockItem { Block = pblock };
 
-                    var blockResult = new BlockResult { Block = pblock };
-                    this.consensusLoop.AcceptBlock(new ContextInformation(blockResult, this.network.Consensus));
-                    this.consensusLoop.Puller.SetLocation(newChain);
+                    this.consensusLoop.AcceptBlock(blockResult);
 
                     if (blockResult.ChainedBlock == null)
-                        break; // Reorg.
+                    {
+                        this.logger.LogTrace("(-)[REORG-2]");
+                        return blocks;
+                    }
 
                     if (blockResult.Error != null)
+                    {
+                        if (blockResult.Error == ConsensusErrors.InvalidPrevTip)
+                            continue;
+
+                        this.logger.LogTrace("(-)[ACCEPT_BLOCK_ERROR]");
                         return blocks;
+                    }
 
                     // Push the block to disk, so it is available when peers ask for it.
                     this.blockRepository.PutAsync(blockResult.ChainedBlock.HashBlock, new List<Block> { pblock }).GetAwaiter().GetResult();
```

Comment:
chmm, in both cases - POS and POW, I can see you have removed a lot of that stuff that cares about having a block being inserted into the system and you only left AcceptBlock - which is doing all that stuff; but I can see you have also left this PutAsync of blockRepository. Is this something that we don't want to be part of AcceptBlock?

Reply:
Yeah PutAsync is part of the store feature (we actually don't have access to it in consensus) the Signal will do that anyway but just to be sure I left the code at the miner, though this can be remove as well.